index.js

```diff
@@ -3,26 +3,101 @@
 'use strict';
 var spawn = require('child_process').spawn;
 
-function potentialExit (childCmd, code) {
+var sh, shFlag, children, args, wait, cmds, verbose, i ,len;
+// parsing argv
+cmds = [];
+args = process.argv.slice(2);
+for (i = 0, len = args.length; i < len; i++) {
+    if (args[i][0] === '-') {
+        switch (args[i]) {
+            case '-w':
+            case '--wait':
+                wait = true;
+                break;
+            case '-v':
+            case '--verbose':
+                verbose = true;
+                break;
+            case '-h':
+            case '--help':
+                console.log('-h, --help         output usage information');
+                console.log('-v, --verbose      verbose logging')
+                console.log('-w, --wait         will not close silbling processes on error')
+                process.exit();
+                break;
+        }
+    } else {
+        cmds.push(args[i]);
+    }
+}
+
+// called on close of a child process
+function childClose (code) {
+    var i, len;
     code = code? (code.code || code) : code;
-    if (code > 0) {
-        console.error('`' + childCmd + '` failed with exit code ' + code);
-        process.exit(code);
+    if (verbose) {
+        if (code > 0) {
+            console.error('`' + this.cmd + '` failed with exit code ' + code);
+        } else {
+            console.log('`' + this.cmd + '` ended successfully');
+        }
+    }
+    if (code > 0 && !wait) close();
```

Comment:
Need to pass `code` to `close()` here - so that `close()` can exit with a non-zero code if the child process did.